File: Shared/Response.cs
using System;
using Newtonsoft.Json;

namespace SignalRCommunicator
{
    [Serializable]
    public class Response : IResponse
    {
        [JsonProperty]
        public string CorrelationId { get; set; }

        [JsonProperty]
        public string Topic { get; set; }

        [JsonProperty]
        public string Content { get; set; }

        [JsonProperty]
        public string Sender { get; set; }

        [JsonProperty]
        public DateTime Timestamp { get; set; }
        
        [JsonProperty]
        public bool Success { get; set; }

        public Response()
        {
            CorrelationId = Guid.NewGuid().ToString();
        }
        public Response( 
            bool success,
            string correlationId,
            string topic,
            string content,
            string sender,
            DateTime timestamp
            ) :this()
        {
            CorrelationId = correlationId;
            Topic = topic;
            Content = content;
            Sender = sender;
            Timestamp = timestamp;
            Success = success;
        }

        public Response(string topic, bool success)
        {
            Topic = topic;
            Success = success;
        }
        public Response(string topic, string content, bool success)
            : this(topic, success)
        {
            Content = content;
        }

        public Response(Message message, bool success)
            : this(success, message.CorrelationId, message.Topic, message.Content, message.Sender, message.Timestamp)
        {

        }


        /*public Response( string content, string sender, DateTime timestamp, bool success ) : this()
        {
            Content = content;
            Sender = sender;
            Timestamp = timestamp;
            Success = success;
        }*/
    }
}

File: WebClient/src/Response.ts
import { IResponse } from "./IResponse";

export class Response implements IResponse {

    private correlationId: string;
    private content: string;
    private sender: string;
    private topic: string;
    private timeStamp: Date;
    private success: boolean;

    constructor(id: string, content: string, sender: string, topic: string, success: boolean) {
        this.correlationId = id;
        this.content = content;
        this.sender = sender;
        this.topic = topic;
        this.timeStamp = new Date();
        this.success = success;
    }

    public get Topic() { return this.topic; }
    public set Topic(value: string) { this.topic = value; }

    get CorrelationId() { return this.correlationId; }
    set CorrelationId(value: string) { this.correlationId = value; }

    get Sender() { return this.sender; }
    set Sender(value: string) { this.sender = value; }

    get Content() { return this.content; }
    set Content(value: string) { this.content = value; }

    get TimeStamp() { return this.timeStamp; }
    set TimeStamp(value: Date) { this.timeStamp = value; }

    get Success() { return this.success; }
    set Success(value: boolean) { this.success = value; }

}